java/vector/src/main/java/org/apache/arrow/vector/types/pojo/Field.java

```diff
@@ -121,7 +124,18 @@ public static Field convertField(org.apache.arrow.flatbuf.Field field) {
     }
     ImmutableList.Builder<Field> childrenBuilder = ImmutableList.builder();
     for (int i = 0; i < field.childrenLength(); i++) {
-      childrenBuilder.add(convertField(field.children(i)));
+      Field childField = convertField(field.children(i));
+      if ((field.typeType() == Type.List || field.typeType() == Type.FixedSizeList)
+        && childField.getName().equals(ZeroVector.NAME)) {
+        Field modifiedField = new Field(DATA_VECTOR_NAME, childField.isNullable(), childField.getType(),
+          childField.getDictionary(),
+          childField.getChildren(),
+          childField.getTypeLayout(),
+          childField.getMetadata());
+        childrenBuilder.add(modifiedField);
+        continue;
+      }
+      childrenBuilder.add(childField);
```

Comment:
Could you put this block in a separate method with a name that makes it obvious what it is changing and why?

Reply:
sure

Reply:
Any other comments?